### PR TITLE
Add aci_vmm_credential.py

### PR DESCRIPTION
**Pull request by jasonjuenger**
##### SUMMARY
Adds a new network module, aci_vmm_credential, to create credential objects for VMM integrations. This module partially fulfills feature request #34885.

##### ISSUE TYPE
- New Module Pull Request

##### COMPONENT NAME
aci_vmm_credential.py

### CODE REVIEW

**Comment by jasonjuenger** [comment]
This module was tested successfully on ACI version 4.1(1j).

File: lib/ansible/modules/network/aci/aci_vmm_credential.py [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(
+        name=dict(type='str', aliases=['credential_name', 'credential_profile']),
+        credential_password=dict(type='str'),
+        credential_username=dict(type='str'),
+        description=dict(type='str', aliases=['descr']),
+        domain=dict(type='str', aliases=['domain_name', 'domain_profile']),
+        state=dict(type='str', default='present', choices=['absent', 'present', 'query']),
+        vm_provider=dict(type='str', choices=['cloudfoundry', 'kubernetes', 'microsoft', 'openshift', 'openstack', 'redhat', 'vmware'])
```
Comment (Qalthos) [inline lib/ansible/modules/network/aci/aci_vmm_credential.py]: I would probably do something like this, to make sure the choices always match what you can accept.
```suggestion
        vm_provider=dict(type='str', choices=VM_PROVIDER_MAPPING.keys())
```

Comment (jasonjuenger) [reply lib/ansible/modules/network/aci/aci_vmm_credential.py]: Thanks for the recommendation @Qalthos. I made the modification and re-ran integration tests successfully.

**Comment by dagwieers** [comment]
@rsmeyers @devarshishah3 Please advise on naming/functionality.

**Comment by rsmeyers** [comment]
aci_vmm_credential is good

**Pull request merged**